.github/workflows/pythonapp.yml

```diff
@@ -4,35 +4,45 @@ on: [push, pull_request]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
+    timeout-minutes: 60
+
     env:
       DISPLAY: ':99.0'
+
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
+
     - name: Set up Python 3.8
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v3
       with:
         python-version: 3.8
+
     - name: Install plastimatch
       run: |
        sudo apt-get install plastimatch
+
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
+        python -m pip install --upgrade pip wheel setuptools
         pip install -r pre-requirements.txt
         pip install -r requirements.txt
         pip install pytest-xvfb pytest-qt
```

Comment:
Not your changes, but any reason why we install these multiple times?

Reply:
no idea to be honest, should consider looking into consolidation of install commands

Reply:
Changes to install commands tested and works fine, merged into this PR